### PR TITLE
chore: fix lint-staged to run on each commit

### PR DESCRIPTION
**Pull request by bhongy**
Currently, `lint-staged` is not running. This PR fixes husky and lint-staged configuration, so `lint-staged` runs for each commit.

Details:
- manage dev tooling (`husky`, `lint-staged`, `sort-package-json`, `prettier`) centrally at the repo root for version consistency
- upgrade `husky`, `lint-staged`, `sort-package-json`, `prettier` to the latest versions
- add `.editorconfig`

Note: files that don't change in this PR don't get the lint-staged tasks applied because lint-staged only run on change files (from git). We'll run prettier on the entire codebase after this PR.

### CODE REVIEW

File: packages/formspree-core/package.json [hunk 1]
```diff
@@ -1,8 +1,8 @@
 {
   "name": "@formspree/core",
   "version": "2.8.1",
-  "description": "The core library for Formspree",
```
Comment (bhongy) [inline packages/formspree-core/package.json:4]: change from sort-package-json

File: packages/formspree-react/package.json [hunk 2]
```diff
@@ -1,8 +1,8 @@
 {
   "name": "@formspree/react",
   "version": "2.4.2",
-  "description": "The React component library for Formspree",
```
Comment (bhongy) [inline packages/formspree-react/package.json:4]: change from sort-package-json

File: .husky/pre-commit [hunk 3]
```diff
@@ -0,0 +1,4 @@
+#!/usr/bin/env sh
+. "$(dirname -- "$0")/_/husky.sh"
+
+yarn lint-staged
```
Comment (bhongy) [inline .husky/pre-commit:4]: husky 8 use this https://typicode.github.io/husky/migrating-from-v4.html

File: .lintstagedrc.yaml [hunk 4]
```diff
@@ -0,0 +1,2 @@
+package.json: yarn sort-package-json
+'*.{json,md,yaml}': yarn prettier --write
```
Comment (bhongy) [inline .lintstagedrc.yaml:2]: might need to double-check if these will overlap with package-specific ones causing a race condition https://github.com/okonet/lint-staged#filtering-files

Comment (bhongy) [reply .lintstagedrc.yaml:2]: Running `lint-staged -v -d` and review the result. This works as expected. No overlap.

**Review by colevscode — APPROVED** [review]

**Pull request merged**